src/file_context.rs

```diff
@@ -1,5 +1,46 @@
+use crate::error::Error;
 use std::path::{Path, PathBuf};
 
+
+/// A context manages finding and loading files.
+///
+/// # Example
+/// ```
+/// use std::collections::HashMap;
+/// use rsass::{Context, Error};
+///
+/// #[derive(Clone, Debug)]
+/// struct StaticContext<'a> {
+///     files: HashMap<String, &'a[u8]>,
+/// }
+///
+/// impl<'a> Context for StaticContext<'a> {
+///     type File = &'a [u8];
+///
+///     fn find_file(
+///         &self, name: &str
+///     ) -> Result<Option<(Self, String, Self::File)>, Error> {
+///         if let Some(file) = self.files.get(name).map(|data| *data) {
+///             Ok(Some((self.clone(), name.to_string(), file)))
+///         } else {
+///             Ok(None)
+///         }
+///     }
+/// }
+/// ```
+pub trait Context: Sized + std::fmt::Debug {
```

Comment:
I think the term `Context` by itself is too generic, and even if backed by something other that std::fs, it is still used to resolve sass "files".  So how about calling the trait `FileContext` and the (default) impl `FsFileContext`?  Or maybe the trait should be called `UrlContext`, since the spec notes that: "To ensure that stylesheets work on every operating system, Sass imports files by URL, not by file path. This means you need to use forward slashes, not backslashes, even when you’re on Windows".

Reply:
To me it makes sense to update the names, if strict backwards compatibility is not a requirement.

I prefer your first suggestion, while changing the parameter name to "url"; `FileContext::find_file(&self, url: &str) -> ...<File>`, since they find files by url.

Reply:
Yes, that souds good!  I think next release should be 0.17.0 anyway, so I think good naming and "conceptual" backwards compatible is more important than by-the-compiler backwards compatibility for now.  To get an as nice as possible api before 1.0 and _then_ to stick with it.

Reply:
I agree